more_examples/autosnapshotting/integration_test/app_test.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:autosnapshotting/main.dart' as app;
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+// Run:
+// flutter test integration_test/app_test.dart -d macos
+
+void main() {
```

Comment:
should this test be part of the main leak_tracker/test or part of an example app? It seems like this belongs in testing for the package itself rather than testing for an example app

Reply:
Yes. 
However, there are technical issues to move it. The test uses the folder `macos` in the root of the example.
I prefer to keep technical simplicity.

Reply:
Where is the macos folder used? What is the technical issue that would prevent moving the test where it belongs?

Reply:
Are you referring to the call to `app.main()` that fails? When the test is under leak_tracking/integration_test/, this could an issue with how we are packaging the sample app and including it as a dependency. 

If we leave the test here, can you add a comment at the top explaining that this is not actually and example test that we expect users to duplicate, but rather a way to verify the leak_tracker behavior on a real application.

And then just want to verify one thing - is this test ran on the CI for leak_tracker?

Reply:
macos test is temporary disabled because of flutter issue: https://github.com/flutter/flutter/issues/105913
Added headless test.

About comment for the test you asked for.
The test is actually tests the example is doing what it should do - to demonstrate how auto-snapshotting works. If users modify the example, they need to modify the test.
The same way as for counter application: the autogenerated tests verifies counter is working. And, if users change the app, they should change the test.
How does it sound?